Src/main.c

```diff
@@ -587,14 +591,17 @@ int32_t doPidCalculations(struct fastPID* pidnow, int actual, int target)
 void loadEEpromSettings()
 {
     read_flash_bin(eepromBuffer.buffer, eeprom_address, sizeof(eepromBuffer.buffer));
-
-    if (eepromBuffer.advance_level > 3) {
-        eepromBuffer.advance_level = 2;
+    if(eepromBuffer.eeprom_version != EEPROM_VERSION){
+      NVIC_SystemReset(); // do not load any settings if eeprom not 3
```

Comment:
this makes for a bad user experience in updating firmware. Maybe have some code when going from 2 to 3 to reset just the new variables?
This would also completely break DroneCAN as we'd get into a boot loop as you can't update parameters inside the bootloader